common/lib/capa/capa/capa_problem.py

```diff
@@ -942,11 +942,18 @@ def response_a11y_data(self, response, inputfields, responsetype_id, problem_dat
                 # store <label> tag containing question text to delete
                 # it later otherwise question will be rendered twice
                 element_to_be_deleted = responsetype_label_tag
-            elif 'label' not in inputfields[0].attrib:
+            elif 'label' in inputfields[0].attrib:
+                # Get first <p> tag before responsetype, this <p> may contains the question text.
+                p_tag = response.xpath('preceding-sibling::*[1][self::p]')
```

Comment:
Can we add comment with exact details about picking up first preceding p tag only.